sdk/include/psdk/winbase.h

```diff
@@ -608,10 +608,15 @@ extern "C" {
 
 #define INIT_ONCE_STATIC_INIT RTL_RUN_ONCE_INIT
 
-#if (_WIN32_WINNT >= 0x0600)
+typedef enum _DEP_SYSTEM_POLICY_TYPE {
+    AlwaysOff = 0,
+    AlwaysOn = 1,
+    OptIn = 2,
+    OptOut = 3
+} DEP_SYSTEM_POLICY_TYPE;
+
 #define PROCESS_DEP_ENABLE 0x00000001
```

Comment:
Then I also need to add the Vista ifdef around the functions? In such case, I also will need to set exports to `-version=0x600+`, otherwise it will fail to compile. But in such case, it makes no sense, since the functions will be only compiled for NT6+, when in Windows they are present since XP SP3. So this will be wrong.

Reply:
The functions seem present on XP SP3, but not on server 2003 SP2.

Reply:
So I'd better add XPSP3 ifdef instead.

Reply:
Ah no, it seems to be present in 2003 SP2 also, at least because:
1. MSDN says the following:
>Data Execution Prevention (DEP) is a system-level memory protection feature that is built into the operating system starting with Windows XP **and Windows Server 2003**.

on this page: https://docs.microsoft.com/en-us/windows/win32/memory/data-execution-prevention.
2. In 2003 SP2, the DEP tab in sysdm.cpl also exists, so DEP appears to be included there too.

Reply:
> These functions are required by some apps

Does MiniBrowser 1.0 work on S03sp2?

> The functions seem present on XP SP3, but not on server 2003 SP2.

Indeed!
kernel32.dll, v5.1.2600.5781: 3 names.
kernel32.dll, v5.2.3790.4480: 0 name. (Are these features available differently? Assumed not.)

Then, we may want to add them to kernel32_vista.dll instead...

> https://docs.microsoft.com/en-us/windows/win32/memory/data-execution-prevention

Yeah, meanwhile these 3 function pages do read:
```
Minimum supported client | Windows Vista with SP1, Windows XP with SP3 [desktop apps only]
Minimum supported server | Windows Server 2008 [desktop apps only]
```

Reply:
Sorry, my mistake. These APIs are indeed missing on Windows Server 2003 SP2. :sweat_smile: They exist only on XP SP3. So we indeed should separate this stuff via ifdef, but via XP SP3 ifdef still, because otherwise these APIs will be unavailable for the apps developed for XP, so they will fail.
![dep-w2k3](https://user-images.githubusercontent.com/26385117/141788006-bd6cfbb9-f00c-4c0f-b094-1b3e5d7efd0b.png)

Reply:
> XP SP3 ifdef

MSDN reads`To compile an application that calls this function, define _WIN32_WINNT as 0x0600 or later.`...

Reply:
https://github.com/adobe/chromium/blob/master/sandbox/src/dep.cc#L46
claims:
>   // Only available on Windows XP SP2 and **Windows Server 2003 SP1**.
  // For reference: http://www.uninformed.org/?v=2&a=4

🤔

Reply:
This comment is in a PSDK header, so it doesn't really matter what's exported where. Microsoft's header has the ifdef, so ours should, too. If that causes trouble for our kernel32, then it needs to set `_WIN32_WINNT` to Vista.